src/test/java/htsjdk/samtools/SamReaderSortTest.java

```diff
@@ -93,6 +100,47 @@ public void testSortAssertionFails(String file, SAMFileHeader.SortOrder order) t
         }
     }
 
+    private CRAMFileReader getCramFileReader(String file, String fileReference) {
+        final ReferenceSource referenceSource = new ReferenceSource(new File(fileReference));
+        return new CRAMFileReader(new File(file), referenceSource);
+    }
+
+    @Test(dataProvider = "sortsCramWithoutIndex")
+    public void testCramSort(String file, String fileReference, SAMFileHeader.SortOrder order) throws Exception {
+        final CRAMFileReader cramFileReader = getCramFileReader(file, fileReference);
+        final SAMRecordIterator samRecordIterator = cramFileReader.getIterator().assertSorted(order);
+        while (samRecordIterator.hasNext()) {
+            samRecordIterator.next();
+        }
+    }
```

Comment:
should probably check if any record has been read in case reader failed silently